sourmash_lib/sbt.py

```diff
@@ -376,51 +623,70 @@ def combine(self, other):
 class Node(object):
     "Internal node of SBT."
 
-    def __init__(self, factory, name=None, fullpath=None):
+    def __init__(self, factory, name=None, path=None, storage=None):
         self.name = name
+        self.storage = storage
         self._factory = factory
         self._data = None
-        self._filename = fullpath
+        self._path = path
 
     def __str__(self):
         return '*Node:{name} [occupied: {nb}, fpr: {fpr:.2}]'.format(
                 name=self.name, nb=self.data.n_occupied(),
                 fpr=khmer.calc_expected_collisions(self.data, True, 1.1))
 
-    def save(self, filename):
-        self.data.save(filename)
+    def save(self, path):
+        # We need to do this tempfile dance because khmer only load
+        # data from files.
+        with NamedTemporaryFile(suffix=".gz") as f:
```

Comment:
First example of the 'tempfile dance'. Storages return buffers now, but for `Nodegraph` we need to save/load something from disk. It would be cool to add loading/saving from `StringIO` and similar to khmer =]